zebra-chain/src/orchard_zsa/issuance.rs

```diff
@@ -0,0 +1,260 @@
+//! Orchard ZSA issuance related functionality.
+
+use std::{fmt::Debug, io};
+
+use crate::{
+    block::MAX_BLOCK_BYTES,
+    serialization::{
+        zcash_serialize_empty_list, ReadZcashExt, SerializationError, TrustedPreallocate,
+        ZcashDeserialize, ZcashDeserializeInto, ZcashSerialize,
+    },
+};
+
+use nonempty::NonEmpty;
+
+// FIXME: needed for "as_bool" only - consider to implement as_bool locally
+use bitvec::macros::internal::funty::Fundamental;
+
+use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+
+use orchard::{
+    issuance::{IssueAction, IssueBundle, Signed},
+    keys::IssuanceValidatingKey,
+    note::{RandomSeed, Rho},
+    primitives::redpallas::{SigType, Signature, SpendAuth},
+    value::NoteValue,
+    Address, Note,
+};
+
+use super::serialize::ASSET_BASE_SIZE;
+
+/// Wrapper for `IssueBundle` used in the context of Transaction V6. This allows the implementation of
+/// a Serde serializer for unit tests within this crate.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct IssueData(IssueBundle<Signed>);
+
+// Sizes of the serialized values for types in bytes (used for TrustedPreallocate impls)
+// FIXME: are those values correct (43, 32 etc.)?
+//const ISSUANCE_VALIDATING_KEY_SIZE: u64 = 32;
+const ADDRESS_SIZE: u64 = 43;
+const NULLIFIER_SIZE: u64 = 32;
+const NOTE_VALUE_SIZE: u64 = 4;
+const RANDOM_SEED_SIZE: u64 = 32;
+// FIXME: is this a correct way to calculate (simple sum of sizes of components)?
+const NOTE_SIZE: u64 =
```

Comment:
It says [147 here](https://github.com/zcash/zips/pull/854/files#diff-57dfe6dfbd7af7e21e6f4731a1c1af0028c106e8dd61122cf64d22ffd0474fa9R326) but these add up to 143, this or the ZIP may need an update.